MaxText/accelerator_to_spec_map.py

```diff
@@ -167,7 +167,7 @@ class SystemCharacteristics:
         'tpu', 'v5:4x4x92', 'megacore', (2, 2, 1), 1472
     ),
     'v5p-3072': SystemCharacteristics(
-        'tpu', 'v5:4x12x16', 'megacore', (2, 2, 1), 1536
```

Comment:
rude don't do this. make it a new name.

Reply:
v5p-3072-special or something. If you really want we could optionally provide the topology seprately.

Reply:
This is for AOT needs right? Recall that XPK has its own list.

Reply:
Hi, @rwitten. There was a typo in this topology originated in a g3 source. The original number of devices (1536) doesn't equal to the product of topology i.e. 4x12x16. And this PR is to fix it. 

I have 2 changes in xpk and internal g3 accordingly:
* g3 source cl/609827788
* [xpk PR](https://github.com/google/xpk/pull/76)

cc @Obliviour

Reply:
Agreed to Zhiyu's comment, this fixes the topology that is currently invalid. 

@ZhiyuLi-goog its probably worth adding in the title / description of the intent of the PR to help avoid confusion. Like the total of "Fix invalid topology for v5p-3072"

The description is something like:

Problem:
The previous topology is invalid because...

Solution:
I changed it to be x. Also changed in xpk and ctpu code

Reply:
Thank you both.